### ISSUE TITLE
Changing .config.CLASSIFIER_METRIC doesn't affect .score_model

### ISSUE THREAD

**Issue by thomasfrederikhoeck** [andersbogsnes/ml_tooling#130]
#### Code Sample

```python
from sklearn.datasets import load_iris
from sklearn.ensemble import RandomForestClassifier
from ml_tooling import BaseClassModel

class IrisModel(BaseClassModel):
    def get_training_data(self):
        data = load_iris()
        return pd.DataFrame(data=data.data, columns=data.feature_names), data.target
    
    def get_prediction_data(self):
        pass

rf_clf = IrisModel(RandomForestClassifier(n_estimators=10, random_state=42))
rf_clf.config.CLASSIFIER_METRIC = 'fowlkes_mallows_score'
rf_clf.score_model()
```
results in:
```
[13:21:04] - Scoring model...
[13:21:04] - RandomForestClassifier - accuracy: 0.9736842105263158
<Result RandomForestClassifier: accuracy: 0.97 >
```
#### Problem description

You would expect changing the metric would change the used metric. I'm aware that one can use
 ```python
rf_clf.default_metric = 'fowlkes_mallows_score'
```
but this might not be clear for the user.

Checklist
---------
- [x] I have updated ml-tooling to the newest version `pip install ml-tooling --upgrade`
- [x] I have checked that this issue does not already exist

**Comment by thomasfrederikhoeck** [comment]
I suggest remocing default_metric and instead introduce a helper function checks model type and then looks up in the config.

**Comment by thomasfrederikhoeck** [comment]
@andersbogsnes will make a suggestion.

**Comment by andersbogsnes** [comment]
I agree - we can turn .metric into a property that looks up in the config dynamically - that would probably be cleaner

**Comment by andersbogsnes** [comment]
closed by #132
